lib/reek/smells/smell_detector.rb

```diff
@@ -103,7 +60,6 @@ def exception?(context)
 
       protected
 
-      # NOTE: Needs to be protected so += works for Ruby < 2.2
```

Comment:
That was my bad, it was indeed needed. However I solved it differently so we still get rid of the comment, please see my second commit.